source/internal/_pipeWith.js

```diff
@@ -0,0 +1,27 @@
+import _arity from './_arity';
+import reduce from '../reduce';
+import tail from '../tail';
+
+export default function _pipeWith(transformFn) {
+  return function() {
+    if (arguments.length === 0) {
+      throw new Error('composition must contains at least one function');
+    }
+
+    var pipeArgs = arguments;
+
+    return _arity(
+      arguments[0].length,
+      function() {
+        var inputArgs = arguments;
+
+        return reduce(
+          function(result, f) {
+            return transformFn.call(this, f, result);
+          },
+          pipeArgs[0].apply(this, inputArgs),
+          tail(pipeArgs)
```

Comment:
This is a known [optimization killer](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#32-leaking-arguments).  The `arguments` object is "leaked" (it [escapes](https://en.wikipedia.org/wiki/Escape_analysis)) and slower code is generated.

Reply:
Unfortunately no. It is necessary to copy the arguments in a loop.

Reply:
I rewrite it to copy

Reply:
Due to _arity fn there is always maximum of 10 arguments, so I think the current way should be fine.